src/pythermalcomfort/models.py

```diff
@@ -4633,3 +4643,676 @@ def bmr(self):
+        print(f"During initialization, overall clo value is {self.overall_clo}")
+
+    def _calculate_comfort_setpt(self, overall_clo, par):
+        """Computes and returns the skin temperature when thermally neutral.
+
+        This method runs the JOS-3 model in uniform conditions at a temperature where PMV=0
+        for the provided metabolic rate and clothing insulation values. If the given values match
+        the cached ones, it directly returns the cached comfortable skin temperature set point.
+        Otherwise, it calculates and returns a new comfortable skin temperature set point.
+
+        Parameters
+        ----------
+        overall_clo : float
+            Overall clothing insulation value [clo]
+        par : float
+            Physical activity ratio [-]
+
+        Returns
+        -------
+        comfort_setpt_skin : np.array
+            Skin temperature set point for sensation/comfort model
+
+        Notes
+        -----
+        - If both `clo_overall` and `par` values match the cached values, this function will return
+          the cached comfortable skin temperature set point directly.
+        - The calculation of the comfortable skin temperature set point involves determining the operative temperature
+          where PMV is zero and then running the model in uniform conditions to extract skin temperatures as set points.
+        """
+        # If the clothing insulation value 'clo' and the metabolic rate 'par' are the same as the cached values,
+        # return the cached comfortable skin temperature set point
+        if overall_clo == self.cached_overall_clo and par == self.cached_par:
+            # print("Both 'clo' and 'par' are the same as the cached values. Cached values are applied to comfort set-points")
+            return self.comfort_setpt_skin
+
+        # Otherwise, calculate a new comfortable skin temperature set point
+        else:
+            # Calculate the metabolic rate based on the provided 'par' value.
+            min_met = 1.0
+            met = max(self.bmr * self.par / 58.15, min_met)
+
+            # Calculate the metabolic rate based on the provided 'par' value.
+            overall_clo = np.average(self._clo, weights=Default.local_bsa)
+
+            # Cash PAR (physical activity ratio) and overall clothing insulation
+            self.cached_par = self.par
+            self.cached_overall_clo = self.overall_clo
+
+            # Use the parent class's method to calculate the operative temperature where PMV is zero.
+            self.to = self._calculate_operative_temp_when_pmv_is_zero(
+                met=met, clo=overall_clo
+            )
+
+            # Run the model in uniform conditions at a temperature where PMV=0
+            # for the given met and clo and use the resulting skin temperatures as the setpoints
+            # Assuming that the initial conditions are uniform and at a temperature where PMV=0
+            print("Skin neutral temperature calculation starts")
```

Comment:
should we consider using LOGs rather than prints?

Reply:
Thank you for your advise! That is the way!